### ISSUE TITLE
Fix: use feature flag for plugin system

### ISSUE THREAD

**Issue by Luxzi** [lthoerner/rush#47]
Currently the plugin system brings in the most dependencies and most of those dependencies are only used there furthermore they greatly increase compile times making rapid development more tedious. **The plugin system should be put behind a feature flag**. This issue is not severe but would make development much easier in other non-plugin based parts of the code base as compile times would greatly improve.

*Cargo.toml*
```
[features]
default  = []
plugin-system = []
```

*Any file that implements the system*
```
#[cfg(feature = "plugin-system"]
fn your_function here() {}
```

**Comment by doinkythederp** [comment]
Yeah at lea at until it's somewhat useful this is a good idea

**Comment by lthoerner** [comment]
Fixed via #48
